Pocket/pocket-ios
# My List: Show skeleton cells during intial download

## Summary
Show skeleton cells during intial download

## References 
https://getpocket.atlassian.net/browse/IN-570

## Implementation Details
In order to keep track of the current state of the initial download, we add a `initialDownloadState` property to `Source`. This property contains an instance of the enum `InitialDownloadState` that can be one of `.unknown`, `.started`, `paginating`, or `.completed`.
This allows us to respond appropriately when determining how to build a snapshot for My List. For example, when building the snapshot while the download state is `.started`, we know to simply include a few skeleton cells to signal to the user that content is being fetched. Once we start `paginating` we can use the `totalCount` associated value to determine the number of placeholder cells that need to be rendered alongside the actual content that has been loaded.

## Test Steps
1. Launch app for first time, navigate to "My List/My List"
2. Before any content has been loaded, you should see 4 placeholder/skeleton cells3. 
3. After content starts to load, scroll down quickly
4. Observe that there are placeholder cells that will transition to cells with actual content as the pagination process continues

## Checklist:

## Screenshots
### Initial load
![initial-load](https://user-images.githubusercontent.com/323541/179869237-63ec5505-e11b-4c4e-9231-23dc097db858.png)

### Paginating
![paginating](https://user-images.githubusercontent.com/323541/179869242-42980e58-263c-4af9-9279-0e0e90fd2ee1.png)
